valhalla/common/telescope_states.py

```diff
@@ -15,116 +15,188 @@
-                        },
-                        {
-                            "terms": {
-                                "site": sites
-                            }
-                        }
-                    ]
+                        ]
+                    }
                 }
             }
         }
-    }
-
-    event_data = get_es_data(date_range_query)
-    telescope_status = {}
-    last_event = {}
-
-    for events in event_data:
-        # ignore the enclosure_interlock state as it is redundant
-        if events['_source']['type'].upper() == 'ENCLOSURE_INTERLOCK':
-            continue
-        telescope_key = TelescopeKey(site=events['_source']['site'],
-                                     observatory=events['_source']['enclosure'],
-                                     telescope=events['_source']['telescope'])
-        if telescope_key in available_telescopes:
-            timestamp = datetime.strptime(events['_source']['timestamp'],
-                                          ES_STRING_FORMATTER).replace(tzinfo=timezone.utc)
-            if telescope_key not in telescope_status:
-                telescope_status[telescope_key] = []
-                last_event[telescope_key] = None
-
-            if not last_event[telescope_key]:
-                last_event[telescope_key] = events['_source']
-            elif (last_event[telescope_key]['type'] != events['_source']['type']
-                  or last_event[telescope_key]['reason'] != events['_source']['reason']):
-                telescope_status[telescope_key].append({'telescope': str(telescope_key),
-                                                        'event_type': last_event[telescope_key]['type'],
-                                                        'event_reason': last_event[telescope_key]['reason'],
-                                                        'start': datetime.strptime(
-                                                            last_event[telescope_key]['timestamp'],
-                                                            ES_STRING_FORMATTER).replace(tzinfo=timezone.utc),
-                                                        'end': timestamp})
-                last_event[telescope_key] = events['_source']
-
-    for telescope_key, event in last_event.items():
-        if event:
-            telescope_status[telescope_key].append({'telescope': str(telescope_key),
-                                                    'event_type': event['type'],
-                                                    'event_reason': event['reason'],
-                                                    'start': datetime.strptime(event['timestamp'],
-                                                                               ES_STRING_FORMATTER)
-                                                    .replace(tzinfo=timezone.utc),
-                                                    'end': end})
-
-    return telescope_status
+        try:
```

Comment:
I would prefer this be in the init method of the class, perhaps assigning to self.es